community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -332,21 +343,33 @@ public Transaction beginTx()
         return beginTransaction( KernelTransaction.Type.explicit, AccessMode.Static.FULL );
     }
 
+    @Override
+    public Transaction beginTx( long timeout )
+    {
+        return beginTransaction( KernelTransaction.Type.explicit, AccessMode.Static.FULL, timeout );
+    }
+
     public InternalTransaction beginTransaction( KernelTransaction.Type type, AccessMode accessMode )
     {
-        if ( spi.isInOpenTransaction() )
-        {
-            // FIXME: perhaps we should check that the new type and access mode are compatible with the current tx
-            return new PlaceboTransaction( spi::currentTransaction, spi::currentStatement );
-        }
+        return beginTransaction( () -> spi.beginTransaction( type, accessMode ) );
```

Comment:
Are we sure this won't result in `new` object allocation at runtime? We should generally avoid "creating" lambdas on hot paths unless we're sure what hotspot does to them.